### DISCUSSION TITLE
Example: ModbusServerContext(single=False)

### DISCUSSION THREAD

**Discussion by titov32** [pymodbus-dev/pymodbus#1957] [Q&A]
I need to create multiple device emulators, and it's highly preferable for each of them to have a different context, including both different data and identifying information. The documentation examples only provide ModbusServerContext(single=True), indicating that they will share the same context. Are there examples with single=False to create multiple device emulators with different contexts?
A simple change of the key leads to a startup error

**Comment by janiversen** [comment]
A server can serve multiple slaves by simply defining different data content, BUT the modbus device identification is identical in all slaves because there are only one server instance.

So to have different data/registers simply define multiple slaves.

To have multiple device identification simply start multiple servers.

**Comment by janiversen** [comment]
The server examples have 3 slaves.

**Comment by janiversen** [comment]
We have lately had a couple of requests to allow different identification for each slave, however this is not something currently planned, but we are happy to accept pull requests,

Again different context is standard, just specify multiple slaves and single=false, see examples,

**Reply by titov32** [reply]
What is the potential number of clients for servers? Are there any limitations?

**Reply by janiversen** [reply]
Memory and number of open sockets, but nothing in our code.
